app/src/test/kotlin/io/eugenethedev/taigamobile/repositories/TasksRepositoryTest.kt

```diff
@@ -0,0 +1,826 @@
+            )
+            assertEquals(
+                expected = sprints[index % sprints.size].name,
+                actual = commonTaskAfterChange.sprint?.name
+            )
+        }
+
+        userStories.forEachIndexed { index, story ->
+            tasksRepository.changeSprint(
+                story.id,
+                story.taskType,
+                sprints[index % sprints.size].id,
+                story.version
+            )
+            val storyAfterChange = tasksRepository.getCommonTask(story.id, CommonTaskType.UserStory)
+
+            testCommonTaskExtEquality(story, storyAfterChange)
+            assertEquals(
+                expected = sprints[index % sprints.size].id,
+                actual = storyAfterChange.sprint?.id
+            )
+            assertEquals(
+                expected = sprints[index % sprints.size].name,
+                actual = storyAfterChange.sprint?.name
+            )
+        }
+    }
+
+    @Test
+    fun `test link to epic`() = runBlocking {
+        val epics = tasksRepository.getEpics(1, FiltersData())
+        val userStories = tasksRepository.getAllUserStories()
+
+        epics.forEachIndexed { index, epic ->
+            tasksRepository.linkToEpic(epic.id, userStories[(index + 1) % userStories.size].id)
+            val userStoryLink = tasksRepository.getEpicUserStories(epic.id)
+                .find { it.id == userStories[(index + 1) % userStories.size].id }
+
+            assertEquals(
+                expected = userStories[(index + 1) % userStories.size].title,
+                actual = userStoryLink?.title
+            )
+            assertEquals(
+                expected = userStories[(index + 1) % userStories.size].isClosed,
+                actual = userStoryLink?.isClosed
+            )
+            assertEquals(
+                expected = userStories[(index + 1) % userStories.size].status,
+                actual = userStoryLink?.status
+            )
+        }
+    }
+
+    @Test
+    fun `test unlink to epic`() = runBlocking {
+        val epics = tasksRepository.getEpics(1, FiltersData())
+
+        epics.forEach { epic ->
+            tasksRepository.getEpicUserStories(epic.id).forEach { story ->
+                tasksRepository.unlinkFromEpic(epic.id, story.id)
+                val currentUserStories = tasksRepository.getEpicUserStories(epic.id)
+                assertTrue(
+                    story.id !in currentUserStories.map { it.id }
+                )
+            }
+        }
+    }
+
+
+    @Test
+    fun `test change assignees`() = runBlocking {
+        val users = usersRepository.getTeam()
+        val dataForTest = getCommonTasksExt(
+            listOf(
+                CommonTaskType.Epic,
+                CommonTaskType.Issue,
+                CommonTaskType.Task,
+                CommonTaskType.UserStory
+            )
+        )
+
+        CommonTaskType.values().forEach { type ->
+            dataForTest[type]?.forEachIndexed { index, data ->
```

Comment:
See? You are using it here almost right. But I think you should replace safe call (`?`)  with `!!` - we don't want test to pass if there were no test data, right? (like I've suggested above)